### PR TITLE
feat(frontend): add sol listener

### PR DESCRIPTION
**Pull request by loki344**
# Note

review first: https://github.com/dfinity/oisy-wallet/pull/4035

# Motivation

Implement sol listener for balances. Transactions will follow.

# Changes

Implement sol listener similar to btc listener.

# Tests

unit tests provided

### CODE REVIEW

File: src/frontend/src/sol/services/sol-listener.services.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			data: {
+				data: BigNumber.from(balance),
+				certified
+			}
+		});
+	} else {
+		balancesStore.reset(tokenId);
+	}
+
+	// Transaction handling will be added later
```
Comment (DenysKarmazynDFINITY) [inline src/frontend/src/sol/services/sol-listener.services.ts:35]: Something to consider - after sol wallet worker supports transactions, maybe we can combine the btc and sol listeners into one single service as they seem to be completely identical. Transaction stores and error text from the function below can be passed as params. It obviously only one of many places where networks things can be combined, but in this case it seems to be very low hanging fruit which can spare 150 lines of code for us 🙂

**Review by DenysKarmazynDFINITY — APPROVED** [review]
LGTM, thanks

File: src/frontend/src/sol/services/sol-listener.services.ts (hunk 1 above)
Comment (loki344) [reply src/frontend/src/sol/services/sol-listener.services.ts:35]: Yes, totally agree. Also it is planned to migrate eth to the worker approach, which then could benefit from this shared implementation.

Comment (peterpeterparker) [reply src/frontend/src/sol/services/sol-listener.services.ts:35]: Note that having it separated has advantages too, notably a clear separation of concerns and the assurance that if one network fails, it does not impact the other—this without the need to handle the error.

**Pull request merged**